### ISSUE TITLE
Explorer integration

### ISSUE THREAD

**Issue by shish** [canton7/SyncTrayzor#77]
Right click on folder -> "Share via Syncthing" menu item is visible -> hovering over that menu item, a submenu appears with a bunch of checkboxes for each device that syncthing knows about.

(Possibly this could be an entirely separate thing to SyncTrayzor - but IMO making synctrayzor "easy to use windows integration" gives it a lot more value than being just a systray icon and nothing else)

**Comment by canton7** [comment]
This is a bit of a can of worms, to be honest.

Firstly, there's no way to have checkboxes available in the explorer context menu as you describe. There would have to be a 'Share via Syncthing' option, which would launch / restore SyncTrayzor and cause it to display some dialog, where you can select which devices to share the folder with.

At this point, we're starting to duplicate UI that's already in Syncthing. Let's open the can...

 - What should the folder ID be? Presumably we should let the user specify this.
 - How about versioning? Ignores? Rescan interval? Should we let the user specify these? Or should we pick some defaults and let the user go and edit the folder if they want?
 - We'll need to restart Syncthing. So far I've avoided doing this automatically - restarts can take a little while if you've got loads of large folders, and the user may want to do several configuration changes before doing a single restart to apply all of them. I want to avoid automatically restarting if at all possible.

So, would a better option be to open the 'Add Folder' UI that Syncthing already provides, with the folder path pre-populated? That would be nicer, but also pretty hard - Syncthing's UI is a website, and while I can do evil stuff to manipulate it inside SyncTrayzor, I'm relying on Syncthing implementation details which could change at any time, and break SyncTrayzor (this already happened with the 'Open Folder' button in the UI). 

Lastly, I'm not sure useful this would actually be. My understanding is that the common use-case is that the user adds several folders, then uses those from that point forwards. Syncthing doesn't offer Dropbox-like functionality where a link to a specific file can be generated and handed out to anybody

**Comment by shish** [comment]
> Firstly, there's no way to have checkboxes available in the explorer context menu as you describe.

Gah :(


> What should the folder ID be? Presumably we should let the user specify this.

GUID for the unique ID, and folder name for the non-unique friendly name (syncthing doesn't support this yet, there's a discussion about it: https://forum.syncthing.net/t/friendly-names-for-shared-folders/2477/10 )


> How about versioning? Ignores? Rescan interval? Should we let the user specify these? Or should we pick some defaults and let the user go and edit the folder if they want?

The defaults seem non-harmful, so defaults should be fine


> We'll need to restart Syncthing.

Also gah :(  This seems like an implementation detail of the current version of syncthing rather than a fundamental design problem, at least...


> So, would a better option be to open the 'Add Folder' UI that Syncthing already provides

That's slightly less slick, but seems reasonable, if we can get syncthing to provide a sane "prompt the user to add folder <blah>" API.


> My understanding is that the common use-case is that the user adds several folders, then uses those from that point forwards.

Ad-hoc sharing is indeed uncommon - but ad-hoc sharing is currently a massive pain in the ass to set up. I suspect that if it were as easy as "right click -> share with [person] -> job done", it would become common :)

**Comment by canton7** [comment]
I'm tempted to park this until the discussion around friendly folder names is resolved.

If this does make it in, I'd also be tempted to disable it by default, and allow it to be enabled in the settings, since it is likely to be an uncommon use-case and programs adding themselves to my explorer context menu just pissed me off :)

**Comment by shish** [comment]
Yeah, without friendly folder names this would be annoying to use

Have it as an option in the installer version? That seems to be what most not-rude apps with right click integration do; having it only available in the preferences menu, off by default, is equivalent to not existing for most people v_v

**Comment by jkaberg** [comment]
+1 from me, I like it! :smiley:

**Comment by canton7** [comment]
A better way to do this would be to show the 'Add Folder' dialog, and pre-populate the 'Folder Path' input

**Comment by imthenachoman** [comment]
I wasn't sure if I should open a new issue but I think one thing that would be cool is a context menu in Windows to pause/unpause a folder syncing. For example, if I am about to make a lot of changes to a shared folder, I don't want ST to start syncing so I would like to pause it. Right now I have to open ST and that works but being able to do from the context menu would be easier. 

Even better if SyncTrayzor had a "pause for X minutes/hours" option. I know ST doesn't have it so SyncTrayzor would have to time it.

I know there is now way for explorer context menu to know if a folder is shared or not so the option would show up for all folders and it would be up to the end-user to use it on folders that are being shared.

**Comment by canton7** [comment]
I'd welcome contributions!
